File: pset2/caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <stdlib.h>

// Declare global variables
int key;
string plain;

int main(int argc, string argv[])
{
    // Get user input
    if (argc != 2 || argv[1] <= 0)
    {
        printf("SYNTAX: 'caesar [key]'\n");
        return 1;
    }
    key = atoi(argv[1]);
    plain = get_string("plaintext:");

    // Calculate and print ciphertext
    printf("ciphertext:");
    for (int i = 0, l = strlen(plain); i < l; i++)
    {
        if (plain[i] >= 'a' && plain[i] <= 'z')
        {
            printf("%c", (((plain[i] - 'a' + key) % 26) + 'a'));
        }
        else if (plain[i] >= 'A' && plain[i] <= 'Z')
        {
            printf("%c", (((plain[i] - 'A' + key) % 26) + 'A'));
        }
        else
        {
            printf("%c", plain[i]);
        }

    }
    printf("\n");

    return 0;
}

File: pset6/sentimental/Caesar/caesar.py
from cs50 import *

if len(sys.argv) != 2:
    sys.exit("Syntax: python caesar.py [int]")

key = int(sys.argv[1])
plain = get_string("plaintext:  ")
cipher = ""

for i in plain:
    if i >= 'A' and i <= 'Z':
        cipher += chr(((ord(i) - ord('A') + key) % 26) + ord('A'))
    elif i >= 'a' and i <= 'z':
        cipher += chr(((ord(i) - ord('a') + key) % 26) + ord('a'))
    else:
        cipher += i

print("ciphertext: " + cipher)